src/app/util/attribute-table.cpp

```diff
@@ -46,33 +46,13 @@ using namespace chip;
 EmberAfStatus emberAfWriteAttributeExternal(EndpointId endpoint, ClusterId cluster, AttributeId attributeID, uint8_t * dataPtr,
                                             EmberAfAttributeType dataType)
 {
-    EmberAfAttributeWritePermission extWritePermission =
-        emberAfAllowNetworkWriteAttributeCallback(endpoint, cluster, attributeID, dataPtr, dataType);
```

Comment:
This was supposed to be a place where applications can hook over-the-network writes and allow/deny them, no?  Why are we removing that?  Are we sure applications do not use it?

Reply:
This is the bit where I was unsure ... however I have seen no use of it and it uses an ugly hook of using weak-ptr. I am looking to have some path to uncouple things. If we do have users of this I will see how I can make it pluggable instead (and keep it). 

Absolutely trying to figure out if adding this was a violation of `you ain't gonna need it` or not.

Reply:
Well, it's the ACL that is supposed to be the source of truth for access. Having a random block from a callback, is not something that is expected. We don't have extensive tests to guard against arbitrary behaviors there in cert!

Reply:
ACL cannot implement checks based on the actual value being written, which this can.

Now in theory all such checks should be defined by the spec and hence should be implemented in our cluster code and hence applications should not need to do anything here.  The question is whether this theory is correct.

I personally don't mind removing this hook; the question is whether it breaks anyone.

Reply:
I was basing my removal on what seemed zigbee legacy, specifically the comment bits of:

```
...
 Note:   The ZCL specification does not (as of this
 * writing) specify any permission-level security for writing writeable
 * attributes. As far as the ZCL specification is concerned, if an attribute is
 * writeable, any device that has a link key for the device should be able to
 * write that attribute. Furthermore if an attribute is read only, it should not
 * be written over the air. Thus, if you implement permissions for writing
 * attributes as a feature, you MAY be operating outside the specification. This
 * is unlikely to be a problem for writing read-only attributes, but it may be a
 * problem for attributes that are writeable according to the specification but
 * restricted by the application implementing this callback.
 ```
 
 Which does not seem to apply to matter. However it is also true that someone may have decided to implement this for some reason (not sure why they would though...). I am trying to see if sufficient slack communication will find someone that used it or take silence as an indicator of "not used". Unfortunately I can only prove usage if someone answers, however I cannot prove "never used" ...